### PR TITLE
depr(api): deprecate table `get_column`/`s` in favor of `__getitem__`

### PR DESCRIPTION
**Pull request by cpcloud**
Deprecate `get_column`-style API in favor of `col`/`cols`. Open to other naming suggestions!

### CODE REVIEW

**Comment by kszucs** [comment]
I don't have a strong opinion on this, the change itself looks good of course.

What APIs do other libraries provide for accessing columns?

**Comment by cpcloud** [comment]
pandas: `__getitem__`/`__getattr__`
spark: `col("x")` (top level function)
sqlalchemy: `.c[...]`/`.c.<attr>` `.c` is an alias for `.columns`
polars: `col("x")` (top level function)
datafusion: `column` (top level function)

It's kind of all over the place.

**Comment by jcrist** [comment]
Is there a strong reason to provide a method here rather than `__getitem__`/`__getattr__` (or `.select(...)` for `cols`)? I confess I didn't even know we had a `get_column`/`get_columns` method.

**Comment by cpcloud** [comment]
@jcrist Not particularly. My thought was that `expr.col` is slightly less annoying to read and easier to discover if you want to pass it around versus `expr.__getitem__` or `expr.__getattr__`. No strong opinions on it though.

**Comment by jcrist** [comment]
Yeah, I personally prefer:

```python
col = table.foo
col = table["foo"]
```

to

```python
col = table.col("foo")
```

Less typing, and doesn't add a new set of methods to `Table`  which prevents columns with those names from being accessed as attributes (although a column named `col` may not be a good idea anyway). The `col` method seems redundant for `__getitem__`/`__getattr__`, which are what pandas users will be used to using.

I'm not sure I understand the use case for `get_columns`/`cols` though. When might you want to extract a collection of `Column` objects rather than working with them on the parent table directly? Does that happen often enough that you need a special `get_columns`/`cols` method for it?

Anyway, not a blocker from me. My (slight) preference would be to deprecate `get_column`/`get_columns` in favor of using the existing `__getitem__`/`__getattr__` apis (and maybe a manual list comprehension for when `get_columns` is needed :shrug:).

**Comment by cpcloud** [comment]
That seems fine to me, I'd rather get rid of them too.

**Comment by kszucs** [comment]
> Anyway, not a blocker from me. My (slight) preference would be to deprecate `get_column`/`get_columns` in favor of using the existing `__getitem__`/`__getattr__` apis (and maybe a manual list comprehension for when `get_columns` is needed 🤷).

Agree.

File: ibis/expr/types/relations.py [hunk 1]
```diff
@@ -210,33 +210,39 @@ def _ensure_expr(self, expr):
...
-        ...        ('a', 'int64'),
-        ...        ('b', 'string'),
-        ...        ('c', 'timestamp'),
-        ...        ('d', 'float'),
-        ...    ],
-        ...    name='t'
-        ... )
-        >>> a, b, c = table.get_columns(['a', 'b', 'c'])
+        >>> table = ibis.table(dict(a='int64', b='string', c='timestamp', d='float'))
+        >>> a, b, c = table.cols(['a', 'b', 'c'])
```
Comment (kszucs) [inline ibis/expr/types/relations.py:228]:
```suggestion
        >>> a, b, c = table.get_columns(['a', 'b', 'c'])
```

File: ibis/expr/types/relations.py [hunk 2]
```diff
@@ -210,33 +210,39 @@ def _ensure_expr(self, expr):
         else:
             return expr
 
+    @util.deprecated(
+        version="5.0", instead="use a list comprehension and attribute/getitem syntax"
```
Comment (kszucs) [inline ibis/expr/types/relations.py:214]: Shouldn't this be `4.1`? I'm usually a bit confused the version argument here.

Perhaps we should have two arguments, one highlighting the version we deprecated the API in `asof` and one for marking the version it's going to be removed at?

`util.deprecated(asof="4.1", removed_in="5.0", instead=...)`

Comment (cpcloud) [reply ibis/expr/types/relations.py:214]: Good idea! I'll add those arguments.

Comment (cpcloud) [reply ibis/expr/types/relations.py:214]: For clarity, I've been treating `version` like your `removed_in`.

Comment (kszucs) [reply ibis/expr/types/relations.py]: We can use `removed_in` to raise if we forget to remove the API before the release.

Comment (cpcloud) [reply ibis/expr/types/relations.py]: How so?

**Review by kszucs — APPROVED** [review]
Nice!

**Pull request merged**